test/modern_bpf/test_suites/syscall_exit_suite/accept4_x.cpp

```diff
@@ -1,11 +1,233 @@
 #include "../../event_class/event_class.h"
 
-#ifdef __NR_accept4
+#if defined(__NR_accept4) && defined(__NR_connect) && defined(__NR_socket) && defined(__NR_bind) && defined(__NR_listen) && defined(__NR_close) && defined(__NR_setsockopt) && defined(__NR_shutdown)
 
-/* We follow the same BPF logic of the `accept` so right now there is no need to test all
- * cases like in the `accept` syscall, here we only want to test that the correct BPF program
- * is called.
- */
+/* On `s390x` architectures only `accept4` (`accept` is not defined) is used so we need to test all the cases also here. */
```

Comment:
@Andreagit97 Very very much appreciated! :heart: I will give this a test run in the next days and report the results. Thanks!